handlers.go

```diff
@@ -679,7 +679,26 @@ func (cm *ContentManager) addDatabaseTrackingToContent(ctx context.Context, cont
 	var objects []*Object
 	cset := cid.NewSet()
 
-	err := merkledag.Walk(ctx, func(ctx context.Context, c cid.Cid) ([]*ipld.Link, error) {
+	defer func() {
+		cm.inflightCidsLk.Lock()
+		err = cset.ForEach(func(c cid.Cid) error {
+			v, ok := cm.inflightCids[c]
+			if !ok || v <= 0 {
+				return xerrors.Errorf("cid should be inflight but isn't: %s", c)
```

Comment:
should we just log here and absorb the error? idk if we want to cause the caller to fail if something went funky here

Reply:
yeah, log and absorb the error.

We also need to delete the cid from the map if we are the caller that drops its count to zero.